zokrates_core/src/typed_absy/result_folder.rs

```diff
@@ -813,6 +813,7 @@ pub fn fold_module<'ast, T: Field, F: ResultFolder<'ast, T>>(
             .into_iter()
             .map(|(key, fun)| f.fold_function_symbol(fun).map(|f| (key, f)))
             .collect::<Result<_, _>>()?,
+        constants: p.constants,
```

Comment:
For consistency with the other folder, this should visit the constants as well. Or neither should, because maybe there can be weird side effects in some analysers, and constant inlining runs first anyways